logging/logging.go

```diff
@@ -57,11 +58,37 @@ func DisableLogging() {
 
 // EnableStructuredLogs replaces all logging functions with structured logging
 // variants.
-func EnableStructuredLogs() (func(), error) {
-	logger, err := zap.NewProduction()
-	if err != nil {
-		return func() {}, err
+func EnableStructuredLogs(logDebugStdout bool) (func(), error) {
+	// Define level-handling logic.
+	highPriority := zap.LevelEnablerFunc(func(lvl zapcore.Level) bool {
+		return lvl >= zapcore.ErrorLevel
+	})
+	lowPriority := zap.LevelEnablerFunc(func(lvl zapcore.Level) bool {
+		return lvl < zapcore.ErrorLevel
+	})
+
+	// Lock consoles
+	consoleErrors := zapcore.Lock(os.Stderr)
+
+	var consoleDebugging zapcore.WriteSyncer
+	if logDebugStdout {
+		consoleDebugging = zapcore.Lock(os.Stdout)
+	} else {
```

Comment:
We can simplify this statement and avoid the `else` clause to keep the normal code path at a minimal indentation.

``` go
consoleErrors := zapcore.Lock(os.Stderr)
consoleDebugging := consoleErrors
if logDebugStdout {
  consoleDebugging = zapcore.Lock(os.Stdout)
}
```

See https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow.

Reply:
Thank you for pointing this out, I rarely had a chance to code in go lately.